backend/src/main/java/ai/verta/modeldb/App.java

```diff
@@ -189,7 +189,11 @@ public ServletWebServerFactory servletContainer(final GracefulShutdown gracefulS
 
   @Bean
   public S3Service getS3Service() throws ModelDBException {
-    return new S3Service(System.getProperty(ModelDBConstants.CLOUD_BUCKET_NAME));
+    String bucketName = System.getProperty(ModelDBConstants.CLOUD_BUCKET_NAME);
+    if (bucketName != null && !bucketName.isEmpty()) {
+      return new S3Service(System.getProperty(ModelDBConstants.CLOUD_BUCKET_NAME));
+    }
+    return null;
```

Comment:
In the NFS artifact spring boot server, s3 service bean is not required so it should be set it null.

Reply:
this is not part of backward compatibility. it just fixes the NFS server startup fix.